tests/kernel/threads/thread_init/src/main.c

```diff
@@ -205,6 +205,12 @@ void test_kinit_coop_thread(void)
 /*test case main entry*/
 void test_main(void)
 {
+#ifdef CONFIG_APP_SHARED_MEM
+	/* app shared memory needs to be assigned to the following 2 threads */
+	appmem_add_thread_ztest_dom0(T_KDEFINE_COOP_THREAD);
```

Comment:
It would be nice if we could direct this at toplevel, like we already do with K_THREAD_ACCESS_GRANT()